pkg/util/kubernetes/apiserver/common/common.go

```diff
@@ -34,3 +45,51 @@ func GetMyNamespace() string {
 	log.Warnf("There was an error fetching the namespace from the context, using default")
 	return "default"
 }
+
+// GetOrCreateClusterID generates a cluster ID and persists it to a ConfigMap.
+// It first checks if the CM exists, in which case it uses the ID it contains
+// It thus requires get, create, and update perms on configmaps in the cluster-agent's namespace
+func GetOrCreateClusterID(coreClient corev1.CoreV1Interface) (string, error) {
+	myNS := GetMyNamespace()
+
+	cm, err := coreClient.ConfigMaps(myNS).Get(defaultClusterIDMap, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) == false {
+			log.Errorf("Cannot retrieve ConfigMap %s/%s: %s", myNS, defaultClusterIDMap, err)
```

Comment:
just note that we're logging this twice, here and in `start`

Reply:
What i'm doing there is actually to log something generic, about generating the clusterID in `start` (without logging the error message itself as you noticed), and log the actual error here, with a more precise message about _what_ failed, and the actual error message.

Do you think that's reasonable or do you think i should just return the error and not log here, and log the error in `start`? My fear is that we're missing context about _where_ the failure comes from if i do this.

Reply:
fair enough, thanks for explaining this 👍 
although we might need to use `log.Error` instead of `log. Errorf` in `start` if we're not going to log the error message itself there.

Reply:
👍 that's a good point, updating it